### ISSUE TITLE
Estimation of vector elements outside the tranformation, when only some elements of that vector are transformed, is not allowed

### ISSUE THREAD

**Issue by alistairdunn1** [NIWAFisheriesModelling/CASAL2#409]
**Describe the bug**

When estimating a subset of elements of a vector using a transformation (i.e., YCS using the simplex method), estimation of the elements that are not part of the transofmration is not currently allowed. 

For example, if a model has year classes 2000-2020, and simplex transformation is used to standardise years 2000-2015 using the simplex parameter transformation, we are not allowed to estimate the 2016, 2017, etc., year classes. It would be useful to sometimes estimate year classes outside the standardisation range.

**Release version(s) and/or repository branch(es) affected?**

All versions

**Operating system type and version (and build tools types and versions, if applicable)**

All versions

**Steps to reproduce the bug**

Apply a transformation to YCS using simplex. Attempt to estimate a parameter outside the range supplied to the simplex method. The following error will be generated "There is an @estimate block for <parameter_name> this is not allowed for parameters with a @parameter_transformation block"

**Expected behavior**
Model runs and estimates the parameters required

**Actual behavior**
The error "There is an @estimate block for <parameter_name> this is not allowed for parameters with a @parameter_transformation block" is received

**Screenshots**

**Additional context**

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read the Contributors Manual and the Contributors Guidelines before starting any work).

**Comment by Craig44** [comment]
I think this is getting flagged [here](https://github.com/NIWAFisheriesModelling/CASAL2/blob/ce88c11ec22c598f07a6f38710be054befbbcf49/CASAL2/source/AddressableTransformations/AddressableTransformation.cpp#L341).

I added it because I accidently tried to simultaneously estimate a parameter that had an estimated transformation which causes undefined behaviour. If you change that LOG_FATAL to a LOG_WARNING then you should by pass this for your situation.
